mitxgraders/formulagrader/integralgrader.py

```diff
@@ -149,6 +154,15 @@ def get_limits_and_funcs(self, integrand_str, lower_str, upper_str, dummy_var,
                                       allow_inf=True)
         
         varscope[dummy_var] = (upper + lower) / 2
+        if varscope[dummy_var] == float('inf') or varscope[dummy_var] == -float('inf'):
+            if -float('inf') < lower < float('inf'):
+                varscope[dummy_var] = lower
+            elif -float('inf') < upper < float('inf'):
+                varscope[dummy_var] = upper
+            else:
+                # Try this instead?
+                varscope[dummy_var] = 0
+
```

Comment:
**Requested Change:** Initially, I found this a bit confusing ... why do you need the dummy variable value outside of `eval_sum` / `eval_integrand`. I believe you're doing this to get at the `integrand_used` variables below.

An alternative would be to use `PARSER.parse(formula).variables_used` to get the variables used without actually evaluating. Although, I do not love importing/exporting our global parser instance all over the place. So I just made a PR that exports a `parse` function from expressions.py. See https://github.com/mitodl/mitx-grading-library/pull/294

So this could be:
```python
integrand_used = parse(integrand_str).variables_used
```

Reply:
I just saw:

> I'm guessing this was annoying you...

Yes 😛

Reply:
This is a significantly better approach - no need to evaluate the expression here at all, when all we want is the list of functions used. Implementing now!